### PR TITLE
test schema updates, ensure db user exists after sign-up

### PR DESCRIPTION
**Pull request by acd124**

### CODE REVIEW

File: src/app/(auth)/onboarding/page.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export default async function OnboardingRedirect() {
+  const { userId: clerkId } = auth();
+  if (!clerkId) return redirectToSignIn();
+
+  // fetch guarantees upsert immediately after sign up
+  await usersService.getUserByClerkId(clerkId);
+
+  const profileComplete = await usersService.profileComplete(clerkId);
+  return profileComplete ? redirect("/") : redirect("/profile/edit");
```
Comment (ninest) [inline src/app/(auth)/onboarding/page.tsx:13]: Wouldn't the profile always be incomplete when the account has just been created? Or is this to navigate people back home if they accidentally go to the onboarding page

File: src/app/(main)/hosts/[hostListingId]/tenant-form.tsx [hunk 2]
```diff
@@ -151,12 +150,18 @@ export function TenantForm({ hostListingId, suggestions }: Props) {
           />
           <FormField
             control={form.control}
-            name="duration"
+            name="endTime"
             render={({ field }) => (
               <FormItem>
-                <FormLabel>Duration (days)</FormLabel>
+                <FormLabel>Start time</FormLabel>
```
Comment (ninest) [inline src/app/(main)/hosts/[hostListingId]/tenant-form.tsx]: Should the label be "End time"?

File: src/app/(main)/hosts/[hostListingId]/tenant-schemas.ts [hunk 3]
```diff
@@ -5,6 +5,6 @@ export const tenantRequestFormSchema = z.object({
   itemsDescription: z.string().trim().min(1),
   sqft: z.coerce.number().min(1, "The square footage cannot be zero"),
   startTime: z.coerce.date(),
-  duration: z.coerce.number().min(1, "The duration cannot be 0"),
+  endTime: z.coerce.date(),
```
Comment (ninest) [inline src/app/(main)/hosts/[hostListingId]/tenant-schemas.ts]: We can use [`superRefine`](https://github.com/colinhacks/zod) for validation across more than 1 field (for example, confirming password by having user type it twice, or ensuring that the end time is after the start time in this case)

File: src/app/(auth)/onboarding/page.tsx (hunk 1 above)
Comment (acd124) [reply src/app/(auth)/onboarding/page.tsx:13]: Prevent accidental navigation

File: src/app/(main)/hosts/[hostListingId]/tenant-form.tsx (hunk 2 above)
Comment (acd124) [reply src/app/(main)/hosts/[hostListingId]/tenant-form.tsx]: That was a mistake

File: src/app/(main)/hosts/[hostListingId]/tenant-schemas.ts (hunk 3 above)
Comment (acd124) [reply src/app/(main)/hosts/[hostListingId]/tenant-schemas.ts]: I had an idea that might be possible, but I wasn't sure

**Pull request merged**